daktari/check_utils.py

```diff
@@ -0,0 +1,25 @@
+from typing import Union, Type, Set
+
+from daktari.check import Check
+
+
+class CyclicCheckException(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def get_all_dependent_check_names(check: Union[Check, Type[Check]]) -> Set[str]:
+    return _get_all_dependent_check_names_recursive(check, set())
+
+
+def _get_all_dependent_check_names_recursive(check: Union[Check, Type[Check]], prev_parents: Set[str]) -> Set[str]:
+    prev_parents = prev_parents.union({check.name})
+    dependents = {dep.name for dep in check.depends_on}
+    intersection = dependents.intersection(prev_parents)
+    if len(intersection) > 0:
+        raise CyclicCheckException(f"Check [{intersection.pop()}] has cyclic dependencies")
+
+    sub_dependents = [_get_all_dependent_check_names_recursive(dep, prev_parents) for dep in check.depends_on]
+    flat_sub_dependents = {item for subset in sub_dependents for item in subset}
+    result = flat_sub_dependents.union(dependents)
+    return result
```

Comment:
I don't think so really, because `check_for_cycles` will have to recurse as well and so it felt neater to do it in one go. Like, `check_for_cycles` would end up being basically the same function we have at the moment, except returning True/False I think?

Reply:
I think `check_for_cycles` can be 
```
def check_for_cycles(check: Union[Check, Type[Check]], prev_parents: Set[str]):
  if check.name in prev_parents:
     throw error
  [check_for_cycles(sub_check, prev_parents.union({check.name}) for sub_check in check.depends_on] // for each rather than list comprehension
```

and then getting the dependent check names can go back to being 
```
def get_all_dependent_check_names(check: Union[Check, Type[Check]]) -> List[str]:
    dependents = [get_all_dependent_check_names(dep) for dep in check.depends_on]
    flat_dependents = [item for sublist in dependents for item in sublist]
    return flat_dependents + [check.name]
```
which I think is simpler?